src/index.ts

```diff
@@ -394,9 +478,9 @@ export default class Wallet {
       throw new Error('This is a public key only wallet')
     }
 
-    const v3Params: V3Params = mergeToV3ParamsWithDefaults(opts)
+    const v3Params: V3ParamsStrict = mergeToV3ParamsWithDefaults(opts)
 
-    let kdfParams: PBKDFParams | ScryptKDFParams
+    let kdfParams: KDFParams
     let derivedKey: Buffer
     switch (v3Params.kdf) {
       case KDFFunctions.PBKDF:
```

Comment:
I couldn't comment on the code below, but now that `KDFParams.salt` is a buffer and not converted to string, `kdfparams.salt` could be passed to `pbkdf2Sync` and `scryptsy` instead of `v3Params`.

Reply:
Nice catch, I've made the change.